pages/posts/flight-of-a-byte.tsx

```diff
@@ -0,0 +1,112 @@
+                </p>
+                <p>
+                libp2p pubsub is a powerful tool with algorithms which allow it to scale to many peers and to make sure they all get the pubsub message. You can read more in its documentation, but for this blog post we only need to say that the Wesh group has a libp2p topic name like 
+                &nbsp;<code>
+                /orbitdb/bafyreihpdptqwgpuwu4ob6nusxebmdx6fm6mq5pma5hva6eu5u74pflo7e/2ac4e88c5272f900c76fd36989f9780e5b2c95d75d38fe5bcef0345b34bc4806_wesh_group_messages
+                    </code>
+                &nbsp;a pubsub message to this topic that contains the message log head. The libp2p on <code>client1</code> receives this pubsub message and sends it (code) internally to all processes which are subscribed to the topic. The Wesh application of <code>client1</code> has used ActivateGroup and is therefore subscribed to this topic and receives the pubsub message (code).
+                </p>
+                <h3>IPFS to fetch the message</h3>
+                <p>
+                libp2p pubsub doesn&apos;t care about the meaning of a pubsub message. It only makes sure that all subscribers get it. Remember that the next step is to synchronize the message log, which is handled by the local message store of <code>client1</code>. After checking that it&apos;s not a repeat, the pubsub message is forwarded (code) as an EventPubSubMessage to the message store which receives it (code). This confirms that the type of pubsub message is to update the message log head, and it checks to make sure the head was not already processed. (If <code>client1</code> has already received the same message log head, maybe from another peer, then it doesn&apos;t need to do more processing. This may seem trivial, but it&apos;s important because it&apos;s efficient to exchange these small “head” messages frequently between peers.)  
+                </p>
+                <p>
+                The message store receives (code) the event for a new head which contains the <code>CID</code> of the message. The message (which may be large) is fetched with IPFS, not with libp2p pubsub (which is better for small messages). If you don&apos;t know what a <code>CID</code> is and how IPFS uses it to fetch content, then you can read all about it. Notice that using a <code>CID</code> to fetch the message (code) like “Hello” is a separate process from synchronizing heads. Maybe IPFS has already fetched the same <code>CID</code>, or maybe it can be fetched from another group member that is not <code>client2</code>. Wesh takes advantage of the efficient IPFS algorithms.
+                </p>
```

Comment:
Please include the code links for "receives (code)" and "fetch the message (code)". Also please include the link for "read all about it".